### DISCUSSION TITLE
CLI how to use

### DISCUSSION THREAD

**Discussion by mickdupreez** [d-k-bo/python-makemkv#7] [Q&A]
can someone please explain how to use the CLI feature of this,
i did pip install makemkv[cli]

Usage: pymakemkv COMMAND [OPTIONS]
the command pymakemkv does not work

how to i access this command from windows terminal

love the work guys

**Comment by d-k-bo** [comment]
You need to replace `COMMAND` with one of the subcommands `info` (extract disc info), `backup` (create a full backup) or `mkv` (copy a single title (eg. an episode or the movie) from the disc).  Then you can append options to further modify the behaviour.
So, for example, to dump the disc information to a JSON file you can run
```shell
pymakemkv info --info-file cool-movie.json
```
To copy the main title to a `mkvs/` directory, you can run
```shell
pymakemkv mkv -o mkvs
```
The usage of the CLI mostly follows conventions of other CLI tools on unix-like operating systems.

I just tried to use it on windows (I don't use it too often) and encountered some smaller issues and fixed them. So make sure you update to version `0.2.2`.

It would be great if you shared any issues you have, especially since I don't use the CLI myself.

**Comment by kbaegis** [comment]
Suggestions:
1) enable directory creation corresponding to the DVD title (/path/to/parent/$MOVIE_NAME)
2) the ability to chain multiple backups/mkv together (a loop, auto eject, add disk, push tray back in)

**Reply by d-k-bo** [reply]
Hi, thank your for your suggestions. I think preferences can be very different from person to person, so I want to keep the CLI basic and unopinionated. For more advanced workflows you can use it as a Python library and use it how it fits your needs.
